pkg/config/config_test.go

```diff
@@ -1,62 +1,137 @@
 
 		// when
-		gotCfg, err := NewCommunicationsConfig()
+		gotConfigPaths := config.FromEnvOrFlag()
+
+		// then
+		assert.Equal(t, expConfigPaths, gotConfigPaths)
+	})
 
-		//then
+	t.Run("from CLI flag only", func(t *testing.T) {
+		// given
+		fSet := pflag.NewFlagSet("testing", pflag.ContinueOnError)
+		config.RegisterFlags(fSet)
+		err := fSet.Parse([]string{"--config=configs/first.yaml,configs/second.yaml", "--config", "configs/third.yaml"})
 		require.NoError(t, err)
-		require.NotNil(t, gotCfg)
-		assert.Equal(t, expConfig, *gotCfg)
+
+		// when
+		gotConfigPaths := config.FromEnvOrFlag()
+
+		// then
+		assert.Equal(t, expConfigPaths, gotConfigPaths)
 	})
 
-	t.Run("Load from file and override with environment variables", func(t *testing.T) {
+	t.Run("should honor env variable over the CLI flag", func(t *testing.T) {
 		// given
-		t.Setenv("CONFIG_PATH", "testdata")
+		fSet := pflag.NewFlagSet("testing", pflag.ContinueOnError)
+		config.RegisterFlags(fSet)
 
-		fixToken := fmt.Sprintf("TOKEN_FROM_ENV_%d", time.Now().Unix())
-		t.Setenv("COMMUNICATIONS_SLACK_TOKEN", fixToken)
-		var expConfig Communications
-		loadYAMLFile(t, sampleCommunicationConfig, &expConfig)
-		expConfig.Communications.Slack.Token = fixToken
+		err := fSet.Parse([]string{"--config=configs/from-cli-flag.yaml,configs/from-cli-flag-second.yaml"})
+		require.NoError(t, err)
+
+		t.Setenv("BOTKUBE_CONFIG_PATHS", "configs/first.yaml,configs/second.yaml,configs/third.yaml")
 
 		// when
-		gotCfg, err := NewCommunicationsConfig()
+		gotConfigPaths := config.FromEnvOrFlag()
 
-		//then
-		require.NoError(t, err)
-		require.NotNil(t, gotCfg)
-		assert.Equal(t, expConfig, *gotCfg)
+		// then
+		assert.Equal(t, expConfigPaths, gotConfigPaths)
 	})
 }
 
-func loadYAMLFile(t *testing.T, path string, out interface{}) {
-	t.Helper()
+func TestNormalizeConfigEnvName(t *testing.T) {
+	// given
+	tests := []struct {
+		name            string
+		givenEnvVarName string
+		expYAMLKey      string
+	}{
+		{
+			name:            "env var without any camel keys",
+			givenEnvVarName: "BOTKUBE_SETTINGS_KUBECONFIG",
+			expYAMLKey:      "settings.kubeconfig",
+		},
+		{
+			name:            "env var with a camel key at the end",
+			givenEnvVarName: "BOTKUBE_SETTINGS_METRICS__PORT",
+			expYAMLKey:      "settings.metricsPort",
+		},
+		{
+			name:            "env var with two camel keys at the end",
+			givenEnvVarName: "BOTKUBE_COMMUNICATIONS_SLACK__TOKEN_ID__NAME",
+			expYAMLKey:      "communications.slackToken.idName",
+		},
+		{
+			name:            "env var with a camel key in the middle (2 words)",
+			givenEnvVarName: "BOTKUBE_COMMUNICATIONS_SLACK__TOKEN_ID_NAME",
+			expYAMLKey:      "communications.slackToken.id.name",
+		},
+		{
+			name:            "env var with a camel key in the middle (3 words)",
+			givenEnvVarName: "BOTKUBE_COMMUNICATIONS__SLACK__TOKEN_ID_NAME",
+			expYAMLKey:      "communicationsSlackToken.id.name",
+		},
```

Comment:
Works well 👍 I would add another case:


```go
               {
			name:            "multiple camel keys in the path",
			givenEnvVarName: "BOTKUBE_MY__COMMUNICATIONS_RANDOM__WORD_SLACK__TOKEN_ID_NAME",
			expYAMLKey:      "myCommunications.randomWord.slackToken.id.name",
		},
```

I mean just the overall idea, as I know that my test properties names don't look good 😄